tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc

```diff
@@ -76,7 +76,83 @@ std::set<string> GetDeletedServables(
+//Add a new ServableData for the model version to the the vector of versions to aspired
+void UpdateVersionToAspire(
+    const FileSystemStoragePathSourceConfig::ServableToMonitor& servable,
+    const string version_relative_path,
+    const int version_number,
+    std::vector<ServableData<StoragePath>>* versions) {
+    const ServableId servable_id = {servable.servable_name(), version_number};
+    const string full_path =
+        io::JoinPath(servable.base_path(), version_relative_path);
+    versions->emplace_back(ServableData<StoragePath>(servable_id, full_path));
+}
+
+bool IsValidVersionNumber(const string& version_path, int64 *version_number) {
+  return strings::safe_strto64(version_path.c_str(), version_number);
+}
+
+// Update the servable data to include
+// all the model versions found in the base path as aspired versions.
+bool EmitAllVersions(
+    const FileSystemStoragePathSourceConfig::ServableToMonitor& servable,
+    const std::vector<string>& children,   // represents a list of base-path children from the file system.
+    std::vector<ServableData<StoragePath>>* versions) {
+
+  LOG(INFO) << "Emit all servable versions";
+
+  bool version_found = false;
+  for (int i = 0; i < children.size(); ++i) {
+    // Identify all the versions, among children that can be interpreted as
+    // version numbers.
+    int64 version_number;
+    if (IsValidVersionNumber(children[i], &version_number)) {
+      // Emit all the aspired-versions data.
+      UpdateVersionToAspire(servable, children[i], version_number, versions);
+      version_found = true;
+    }
+  }
+
+  return version_found;
+}
+
+// Update the servable data to include
+// the latest version found in the base path as aspired version.
+bool EmitLatestVersion(
+    const FileSystemStoragePathSourceConfig::ServableToMonitor& servable,
+    const std::vector<string>& children, // represents a list of base-path children from the file system.
+    std::vector<ServableData<StoragePath>>* versions) {
+
+  LOG(INFO) << "Emit latest servable version";
+
+  // Identify the latest version, among children that can be interpreted as
+  // version numbers.
+  int latest_version_child_index = 0;
+  int64 latest_version_number;
+  bool version_found = false;
+  for (int i = 0; i < children.size(); ++i) {
+    int64 version_number;
+    if (!IsValidVersionNumber(children[i], &version_number)) {
+      continue;
+    }
+
+    // Check if this is the largest version number
+    if (latest_version_child_index == 0 || latest_version_number < version_number) {
+      latest_version_child_index = i;
+      latest_version_number = version_number;
+    }
+    version_found = true;
+  }
+
+  // Emit the latest aspired version.
+  if (version_found) {
+    UpdateVersionToAspire(servable, children[latest_version_child_index], latest_version_number, versions);
+  }
+
+  return version_found;
+}
+
+// Like PollFileSystemForConfig(), but for servable
```

Comment:
Please revert to original comment:
// Like PollFileSystemForConfig(), but for a single servable.